core/src/main/java/org/keycloak/util/JWKSUtils.java

```diff
@@ -95,4 +105,47 @@ public static JWK getKeyForUse(JSONWebKeySet keySet, JWK.Use requestedUse) {
 
         return null;
     }
+
+    public static KeyWrapper getKeyWrapper(JWK jwk) {
+        JWKParser parser = JWKParser.create(jwk);
+        if (parser.isKeyTypeSupported(jwk.getKeyType())) {
+            return wrap(jwk, parser);
+        } else {
+            return null;
+        }
+    }
+
+    private static KeyWrapper wrap(JWK jwk, JWKParser parser) {
+        KeyWrapper keyWrapper = new KeyWrapper();
+        keyWrapper.setKid(jwk.getKeyId());
+        if (jwk.getAlgorithm() != null) {
+            keyWrapper.setAlgorithm(jwk.getAlgorithm());
+        }
+        keyWrapper.setType(jwk.getKeyType());
+        keyWrapper.setUse(getKeyUse(jwk.getPublicKeyUse()));
+        keyWrapper.setPublicKey(parser.toPublicKey());
+        return keyWrapper;
+    }
+
+    public static String computeThumbprint(JWK key)  {
+        return computeThumbprint(key, JWK_THUMBPRINT_DEFAULT_HASH_ALGORITHM);
+    }
+
+    public static String computeThumbprint(JWK key, String hashAlg)  {
+        Map<String, String> members = new TreeMap<>();
+        members.put(JWK.KEY_TYPE, key.getKeyType());
+
+        for (String member : JWK_THUMBPRINT_REQUIRED_MEMBERS.get(key.getKeyType())) {
```

Comment:
It seems that this method does not consider lexicographic ordering of [RFC 7638](https://datatracker.ietf.org/doc/html/rfc7638#section-3.2).

Reply:
Actually it does - it is enforced by TreeMap which uses the natural ordering of the keys. Alternatively, we can initially populate `JWK_THUMBPRINT_REQUIRED_MEMBERS` using lexicographic ordering and not use TreeMap for the same; let me know if you consider the latter variant more readable.

Reply:
I see. IMO, it is not needed to put members in lexicographic ordering explicitly for codes' readability. Thank you.